src/gui/DatabaseOpenDialog.cpp

```diff
@@ -80,6 +80,16 @@ DatabaseOpenDialog::DatabaseOpenDialog(QWidget* parent)
     connect(shortcut, &QShortcut::activated, this, [this]() { selectTabOffset(1); });
 }
 
+void DatabaseOpenDialog::showEvent(QShowEvent* event)
+{
+    QDialog::showEvent(event);
+    QTimer::singleShot(100, this, [=] {
+        if (m_view->isOnQuickUnlockScreen() && !m_view->unlockingDatabase()) {
+            m_view->triggerQuickUnlock();
+        }
+    });
+}
+
```

Comment:
I tried debugging it a little and was wondering why nothing is triggering. Turns out, this is only the modal dialog box. If you want to show quick unlock in the main UI, then this needs to go directly into the unlock widget. However, you need to take care not to trigger it immediately on the first showEvent after the database is locked.

Reply:
I only wanted this feature for the modal dialog box since that appears on a user action (browser, Auto-Type, etc) instead of just casually opening or viewing the application.

Reply:
We should definitely have it there, but it would also be handy in the main app if we can find a smart way not to show it accidentally too often. On show might work, on focus would definitely be too often.

Reply:
ShowEvent only triggers between a hide/show call (ie, coming back from tray icon). It would not trigger when coming back from minimize or flipping between windows. I can't think of a way to deduce user intent without annoyance in this case.

Reply:
We can investigate this for 2.8.0